.golangci.yml

```diff
@@ -0,0 +1,47 @@
+run:
+  # timeout for analysis, e.g. 30s, 5m, default is 1m
+  timeout: 5m
+
+  # Skip linting _test.go files
+  tests: false
+
+  # Enables skipping of directories:
+  # vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
+  skip-dirs-use-default: true
+
+  # Allow multiple parallel golangci-lint instances running.
+  # If false (default) - golangci-lint acquires file lock on start.
+  allow-parallel-runners: true
+
+linters:
+  enable:
+    - bodyclose
+    - deadcode
+    - errcheck
+    - gocyclo
+    - gofmt
+    - gosimple
+    - govet
+    - ineffassign
+    - lll
+    - staticcheck
+    - structcheck
+    - typecheck
+    - unused
+    - varcheck
+
+linters-settings:
+  gocyclo:
+    # default is 30.
+    min-complexity: 20
+
+  goimports:
+    # to be edited by the template
+    local-prefixes:
+      "github.com/mendersoftware/#CI_PROJECT_NAME#"
```

Comment:
@oleorhagen This will pass to `goimoprts` the `local ...` flag

Reply:
Yes, but I would remove the `PROJECT_NAME`, and keep only the github/mendersoftware prefix, so that say `mender-artifact` is sorted alongside `mender` as our own project in say the mender-client.

But no strong feelings here ofc. It could just as well be counted as any other 3rd part import, I understand the idea.